db-init.sql

```diff
@@ -0,0 +1,166 @@
+-- PersonCategories
+	-- all the different types a person can be
+
+CREATE TABLE PersonCategories (
+	personCategoryId SERIAL PRIMARY KEY NOT NULL,
+	personCategoryName VARCHAR(255) NOT NULL,
+);
+
+	CREATE UNIQUE INDEX personCategoryName_lower ON PersonCategories(lower(personCategoryName));
+
+	INSERT INTO PersonCategories VALUES (0, 'farmers');
+	INSERT INTO PersonCategories VALUES (1, 'traders');
+	INSERT INTO PersonCategories VALUES (2, 'admins');
+	INSERT INTO PersonCategories VALUES (3, 'monitor');
+
+-- PersonAttributeTypes
+	--  all the different types of attributes a person can have
+
+CREATE TABLE PersonAttributeTypes (
+	attrId SERIAL PRIMARY KEY NOT NULL,
+	attrName VARCHAR(255) NOT NULL
+);
+
+	CREATE UNIQUE INDEX personAttrName_lower ON PersonAttributeTypes(lower(attrName));
+
+	INSERT INTO PersonAttributeTypes VALUES (0, 'username');
+	INSERT INTO PersonAttributeTypes VALUES (1, 'passwordHash');
+	INSERT INTO PersonAttributeTypes VALUES (2, 'paymentFrequency');
+	INSERT INTO PersonAttributeTypes VALUES (3, 'notes');
+
+-- PersonCategoryAttributes
+	-- what types of attributes a category of people have
+
+CREATE TABLE PersonCategoryAttributes (
+	personCategoryId SERIAL REFERENCES PersonCategories(personCategoryId) NOT NULL,
+	attrId SERIAL REFERENCES PersonAttributeTypes(attrId) NOT NULL
+);
+
+	-- farmer
+	INSERT INTO PersonCategoryAttributes VALUES (0, 2); -- paymentFrequency
+	INSERT INTO PersonCategoryAttributes VALUES (0, 3); -- notes
+	-- trader
+	INSERT INTO PersonCategoryAttributes VALUES (1, 0); -- username
+	INSERT INTO PersonCategoryAttributes VALUES (1, 1); -- password
+	-- admin
+	INSERT INTO PersonCategoryAttributes VALUES (2, 0); -- username
+	INSERT INTO PersonCategoryAttributes VALUES (2, 1); -- password
+	-- admin
+	INSERT INTO PersonCategoryAttributes VALUES (3, 0); -- username
+	INSERT INTO PersonCategoryAttributes VALUES (3, 1); -- password
+
+-- Person
+
+CREATE TABLE Persons (
+	personUuid UUID PRIMARY KEY NOT NULL DEFAULT uuid_generate_v4(),
+	firstName VARCHAR(255),
+	middleName VARCHAR(255),
+	lastName VARCHAR(255),
+	phoneNumber VARCHAR(20),
```

Comment:
the phone number fields are optional right? i'm just thinking about whether admins and traders need to input their phone number. so if an admin creates the trader accounts, will he know all his trader's numbers? or can he still create trader accounts without knowing all these fields

Reply:
These are optional, so they don't have a `NOT NULL`, I think this is the safer method for now. Maybe someone will ask us to mandate having phone numbers, we can change that then.